src/java/org/apache/cassandra/repair/state/ValidationState.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.repair.state;
+
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.repair.RepairJobDesc;
+import org.apache.cassandra.utils.Clock;
+
+public class ValidationState extends AbstractState<ValidationState.State>
+{
+    public enum State
+    { START, SENDING_TREES }
+
+    public final Phase phase = new Phase();
+    public final RepairJobDesc desc;
+    public final InetAddressAndPort initiator;
+    public long estimatedPartitions;
+    public long estimatedTotalBytes;
+    public long partitionsProcessed;
+
+    public ValidationState(RepairJobDesc desc, InetAddressAndPort initiator)
+    {
+        super(desc.determanisticId(), State.class);
+        this.desc = desc;
+        this.initiator = initiator;
+    }
+
+    @Override
+    public float getProgress()
+    {
+        int currentState = this.currentState;
+        if (currentState == INIT)
+            return 0.0F;
+        if (currentState == COMPLETE)
+            return 1.0F;
+        if (estimatedPartitions == 0) // mostly to avoid / 0
+            return 0.0f;
+        return Math.min(0.99F, partitionsProcessed / (float) estimatedPartitions);
```

Comment:
not sure if it's useful. The cardinal of enum doesn't represent progress of repair. Sometimes one state can contribute most of the processing time.

Since we have  `"status"` column, `"progress_percentage"` is not needed..

Reply:
I feel that is true for every state but Validation, as that is based off estimated vs processed partitions.

I can look to remove progress as you are right, it doesn't make sense outside of: validation, streaming, etc.

Reply:
removed from each table, but kept in validation.